buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java

```diff
@@ -369,6 +380,18 @@ private static int validateAndCalculateChunkSize(int pageSize, int maxOrder) {
         return chunkSize;
     }
 
+    private static int calculateDefaultMaxOrder(int pageSize) {
```

Comment:
If I read this right, it will find the largest chunk size that pass validation. That could end up using a lot of memory. I think we should have more reserved heuristics beyond, say, 16 MiB chunks or there abouts, such that we try to keep memory usage down beyond that point.

Reply:
@chrisvest  Maybe we can change `io.netty.buffer.PooledByteBufAllocator.MAX_CHUNK_SIZE` to  a smaller value?  16 MiB e.g.?

Reply:
@chrisvest , This `calculateDefaultMaxOrder(int pageSize)`  method is called from the static code block, the `pageSize` param is supposed to be the default value of 8192 KiB in common usage.

Reply:
@chrisvest  If we limit the default max chunk size within 16 MiB, then should we also limit the default max page size within a reasonable size?

Reply:
I wasn't thinking to place a limit beyond what we already do, but rather have the algorithm put fewer pages in the chunks as the page size gets bigger.

Reply:
@chrisvest , to approach this, I think the logic in method `io.netty.buffer.PooledByteBufAllocator.validateAndCalculateChunkSize(int pageSize, int maxOrder)` also needs to be modified?

Reply:
Yes, so we'd basically change the heuristic to find a balance between the two variables.